src/web/app/pages-instructor/instructor-course-enroll-page/instructor-course-enroll-page.component.ts

```diff
@@ -94,59 +99,129 @@ export class InstructorCourseEnrollPageComponent implements OnInit {
-              '' : row[hotInstanceColHeaders.indexOf(this.colHeaders[3])].trim(),
-          comments: row[hotInstanceColHeaders.indexOf(this.colHeaders[4])] === null ?
-              '' : row[hotInstanceColHeaders.indexOf(this.colHeaders[4])].trim(),
-        })));
-
-    this.studentService.enrollStudents(
-        this.courseId, studentsEnrollRequest,
-    ).pipe(finalize(() => this.isEnrolling = false)).subscribe((resp: Students) => {
-      const enrolledStudents: Student[] = resp.students;
-      this.showEnrollResults = true;
-      this.statusMessage.pop(); // removes any existing error status message
-      this.statusMessageService.showSuccessToast('Enrollment successful. Summary given below.');
-      this.enrollResultPanelList =
-          this.populateEnrollResultPanelList(this.existingStudents, enrolledStudents,
-              studentsEnrollRequest.studentEnrollRequests);
-    }, (resp: ErrorMessageOutput) => {
-      this.enrollErrorMessage = resp.error.message;
-    }, () => {
-      this.studentService.getStudentsFromCourse({ courseId: this.courseId }).subscribe((resp: Students) => {
-        this.existingStudents = resp.students;
-        if (!this.isExistingStudentsPanelCollapsed) {
-          const existingStudentTable: Handsontable = this.hotRegisterer.getInstance(this.existingStudentsHOT);
-          this.loadExistingStudentsData(existingStudentTable, this.existingStudents);
-        }
-        this.isExistingStudentsPresent = true;
-      });
+        .forEach((row: string[]) => {
+          currentStudentChunk.push({
+            section: row[hotInstanceColHeaders.indexOf(this.colHeaders[0])] === null ?
+                '' : row[hotInstanceColHeaders.indexOf(this.colHeaders[0])].trim(),
+            team: row[hotInstanceColHeaders.indexOf(this.colHeaders[1])] === null ?
+                '' : row[hotInstanceColHeaders.indexOf(this.colHeaders[1])].trim(),
+            name: row[hotInstanceColHeaders.indexOf(this.colHeaders[2])] === null ?
+                '' : row[hotInstanceColHeaders.indexOf(this.colHeaders[2])].trim(),
+            email: row[hotInstanceColHeaders.indexOf(this.colHeaders[3])] === null ?
+                '' : row[hotInstanceColHeaders.indexOf(this.colHeaders[3])].trim(),
+            comments: row[hotInstanceColHeaders.indexOf(this.colHeaders[4])] === null ?
+                '' : row[hotInstanceColHeaders.indexOf(this.colHeaders[4])].trim(),
+          });
+          if (currentStudentChunk.length >= this.numberOfStudentsPerRequest) {
+            this.allStudentChunks.push(currentStudentChunk);
+            currentStudentChunk = [];
+          }
+        });
+    if (currentStudentChunk.length > 0) {
+      this.allStudentChunks.push(currentStudentChunk);
+      currentStudentChunk = [];
+    }
+
+    const enrolledStudents: Student[] = [];
+    this.remainingStudentChunks = [...this.allStudentChunks];
+
+    const enrollRequests: Observable<Students> = concat(
```

Comment:
Maybe can put a comment here to notify the future maintainer why this is done concat: we cannot afford to parallelise with forkJoin when there's data dependency.